crypto/src/key/mod.rs

```diff
@@ -43,10 +48,18 @@ pub enum SignatureError {
 pub enum KeyKind {
     #[codec(index = 0)]
     Secp256k1Schnorr,
+    // Index 1 is used for extended Secp256k1Schnorr
```

Comment:
I don't know why we're doing this. These don't have to be mutually exclusive. In fact, it's not really possible to encode/decode one into the other. Usually, when you decode bytes into whole objects, you use `decode_all`, which will guarantee that the whole object is filled.

Please don't use this unnecessary and confusing limitation. The only relationship between KeyKind and ExtendedKeyKind is that ExtendedKey can be converted into Key. There's no other hidden relationships here. We don't want the reader of the code to be solving puzzles.

Reply:
I have reverted this have the same index for both extended and normal keys. The rationale is that by having a different "version" number it is clear which key is which.

Reply:
While that sounds nice, the problem becomes consistency in the future if we ever add something else. The size should be enough, even though it's not perfect in a long stream.